Assets/Tests/Runtime/ConnectionManagementTests.cs

```diff
@@ -0,0 +1,514 @@
+            }
+
+            Assert.AreEqual(NumberOfClients - 1, nbLoggedInAgainMsgsReceived, "Not all clients received a LoggedInAgain message.");
+            subscriptions.Dispose();
+        }
+
+        [UnityTest]
+        public IEnumerator ClientConnectingWithoutHost_ConnectionFailed()
+        {
+            SetUniqueProfilesForEachClient();
+
+            for (var i = 0; i < NumberOfClients; i++)
+            {
+                LogAssert.Expect(LogType.Error, k_FailedToConnectToServerErrorMessage);
+            }
+
+            yield return ConnectClients();
+
+            for (var i = 0; i < NumberOfClients; i++)
+            {
+                Assert.IsFalse(m_ClientNetworkManagers[i].IsConnectedClient, $"Client{i} is connected while no server is running.");
+            }
+        }
+
+        [UnityTest]
+        public IEnumerator UnexpectedClientDisconnect_ClientReconnectingSuccessfully()
+        {
+            StartHost();
+            AssertHostIsListening();
+
+            SetUniqueProfilesForEachClient();
+
+            yield return ConnectClients();
+            AssertAllClientsAreConnected();
+
+            // wait for client synchronization to be over server-side before shutting down client
+            yield return null;
+            yield return null;
+
+            var nbReconnectingMsgsReceived = 0;
+            var subscriptions = new DisposableGroup();
+
+            subscriptions.Add(m_ClientScopes[0].Container.Resolve<ISubscriber<ConnectStatus>>().Subscribe(message =>
+            {
+                // ignoring the first success message that is in the buffer
+                if (message != ConnectStatus.Success)
+                {
+                    Assert.AreEqual(ConnectStatus.Reconnecting, message, "Received unexpected ConnectStatus message.");
+                    nbReconnectingMsgsReceived++;
+                }
+            }));
+
+            // Disconnecting the client at the transport level
+            m_ClientNetworkManagers[0].NetworkConfig.NetworkTransport.DisconnectLocalClient();
+
+            // Waiting until shutdown is complete
+            yield return new WaitWhile(() => m_ClientNetworkManagers[0].ShutdownInProgress);
+            Assert.IsFalse(m_ClientNetworkManagers[0].IsConnectedClient, "Client0 has not shut down properly.");
+
+            // Waiting for client to automatically reconnect
+            yield return WaitForClientsConnectedOrTimeOut();
+            Assert.IsTrue(m_ClientNetworkManagers[0].IsConnectedClient, "Client0 failed to reconnect.");
+
+            Assert.AreEqual(1, nbReconnectingMsgsReceived, "No Reconnecting message received.");
+            subscriptions.Dispose();
+        }
+
+
+        [UnityTest, Ignore("Test fails because shutdowns do not invoke OnClientDisconnect on the host, so the ConnectionManager doesn't properly transition to the Offline state")]
```

Comment:
If this is a Netcode bug that is tracked, we should track on our end to come back to this test and re-enable.

Reply:
Create a task to come back to this and we're gtg!